modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteTypeCoercion.java

```diff
@@ -271,6 +271,12 @@ protected boolean needToCast(SqlValidatorScope scope, SqlNode node, RelDataType
             if (fromType == null) {
                 return false;
             }
+
+            if (fromType.getSqlTypeName() == SqlTypeName.BIGINT && toType.getSqlTypeName() == SqlTypeName.BIGINT) {
```

Comment:
As mentioned earlier this is not necessary as well because it does not fix a problem with BIGINT literals.

Reply:
It more or less looks like an attempt to fix BIGINTs for scalar subqueries in DML statements. But it is not correct (see the next comment on cast from `ROW` type).